bluetooth: controller: Connection meta property support in LLL

mtpr-ot opened the pull request:
Added support for vendor specific meta data in LLL conn object. This
enables vendors to add state data to connection, for supporting
specialized BLE slave features.

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot commented:
Feature config- and/or type naming may need to change to be in line with conventions (?)

wopu-ot commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -28,6 +28,9 @@
 #include "hal/ccm.h"
 #include "ll_sw/pdu.h"
 #include "ll_sw/lll.h"
+#if defined(CONFIG_BT_CTLR_CONN_PROP_META)
```
Having same guard here and inside the header file seems to be redundant.

mtpr-ot replied:
> Having same guard here and inside the header file seems to be redundant.

OK, removing conditional #include

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c:
```diff
@@ -18,9 +18,16 @@
 #include "lll_conn.h"
 #include "ull_conn_internal.h"
 
+#ifndef RADIO_TXP_DEFAULT
```
This seems to be an unrelated change. Move to a different PR?

mtpr-ot replied:
> This seems to be an unrelated change. Move to a different PR?

Moved to a separate commit. Necessary for compiling the feature.

cvinayak replied:
Move this to top level subsys/bluetooth/controller/hal/radio.h

wopu-ot commented on subsys/bluetooth/controller/ll_sw/ull_adv.c:
```diff
@@ -605,6 +606,9 @@ u8_t ll_adv_enable(u8_t enable)
 		conn_lll->slave.window_size_prepare_us = 0;
 		/* FIXME: END: Move to ULL? */
+#if defined(CONFIG_BT_CTLR_CONN_PROP_META)
+		memset(&conn_lll->prop_meta, 0, sizeof(conn_lll->prop_meta));
```
Maybe this should call a function to initialize the meta data instead of assuming that zero-initialization is the right thing to do.

mtpr-ot replied:
> Maybe this should call a function to initialize the meta data instead of assuming that zero-initialization is the right thing to do.

I would prefer to avoid having vendor specific meta property functions as well. You can always structure your meta data such that setting all to '0' will be an unambiguous operation.

wopu-ot reviewed (approved):
LGTM

carlescufi commented:
@cvinayak could you review please?

cvinayak commented on subsys/bluetooth/controller/Kconfig:
```diff
@@ -535,6 +535,13 @@ config BT_CTLR_LOW_LAT_ULL
 	  Low latency ULL implementation that uses tailchaining instead of while
 	  loop to demux rx messages from LLL.
 
+config BT_CTLR_CONN_PROP_META
```
BT_CTLR_CONN_META, drop proprietary.

cvinayak commented on subsys/bluetooth/controller/hci/hci.c:
```diff
@@ -28,6 +28,9 @@
 #include "ll_sw/pdu.h"
 #include "ll_sw/lll.h"
+#if defined(CONFIG_BT_LL_SW_SPLIT)
+#include "lll_conn_ext.h"
```
include inside lll_conn.h, as it depends on lll_conn_ext.h

cvinayak commented on subsys/bluetooth/controller/ll_sw/ll_tx_pwr.c:
```diff
@@ -18,9 +18,16 @@
 
 #include "lll.h"
+#if defined(CONFIG_BT_LL_SW_SPLIT)
+#include "lll_conn_ext.h"
```
Same here, love lll_conn_ext.h into lll_conn.h

cvinayak commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll_conn_ext.h:
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2018-2019 Nordic Semiconductor ASA
```
Fix copyright

mtpr-ot committed:
bluetooth: controller: Connection meta property support in LLL

Added support for vendor specific meta data in LLL conn object. This
enables vendors to add state data to connection, for supporting
specialized BLE slave features.

Signed-off-by: Morten Priess <mtpr@oticon.com>

cvinayak reviewed (approved):
LGTM
